com/bl/ttt/tictactoe.py

```diff
@@ -49,15 +50,23 @@ def get_previous_turn_player(self):
         else:
             return self.player1
 
-    def get_player_move_log(self):
-        return self.player_move_log
+    def get_player_move_log(self, player_id='both'):
+        if player_id == 'both':
+            x_move_log = list(self.player_move_log.get('X'))
+            o_move_log = list(self.player_move_log.get('O'))
+            players_move_tuples = list(itertools.zip_longest(x_move_log, o_move_log))
+            players_move_list = list(itertools.chain(*players_move_tuples))
```

Comment:
I like that !

Reply:
I had a much clumsier/denser version which I had to hand roll: 
`
l = itertools.zip_longest( self.player_move_log['X'], self.player_move_log['O'] )
return [item for sublist in l for item in sublist if(item is not None)]
`

Reply:
For comprehensions are really pretty fast but nested ones become hard to read unless you split them over several lines

Reply:
Roger that.